decidim-meetings/app/controllers/decidim/meetings/meetings_controller.rb

```diff
@@ -8,17 +8,44 @@ module Meetings
     # Note that it inherits from `Decidim::Features::BaseController`, which
     # override its layout and provide all kinds of useful methods.
     class MeetingsController < Decidim::Meetings::ApplicationController
-      helper_method :meetings, :meeting
+      helper_method :meetings, :meeting, :search_params, :filter
+
+      def index
+        respond_to do |format|
+          format.html
+          format.js
+        end
+      end
 
       private
 
       def meetings
-        @meetings ||= Meeting.where(decidim_feature_id: current_feature.id).order(start_time: :asc)
+        @meetings ||= MeetingsSearch.new(search_params.merge(feature_id: current_feature.id)).results
       end
 
       def meeting
         @meeting ||= meetings.find(params[:id])
       end
+
+      def search_params
+        default_search_params
+          .merge(params.to_unsafe_h.except(:filter))
+          .merge(params.to_unsafe_h[:filter] || {})
+      end
+
+      def filter
+        @filter ||= filter_klass.new(params.dig(:filter, :category_id), params[:order_start_time], params[:scope_id])
```

Comment:
In order to use the rails form helpers, I need this to be wrapped in an object so that the checkboxes and radio buttons get automatically checked and so, based on the URL params. 

Basically, this class wraps the URL params in an object so that the form is simpler to deal with.